content/webapp/components/IIIFViewer/MainViewer.tsx

```diff
@@ -211,16 +210,17 @@ function getPositionData({
   return highlightsPositioningData || [];
 }
 const ItemRenderer = memo(({ style, index, data }: ItemRendererProps) => {
-  const { scrollVelocity, canvases, externalAccessService } = data;
+  const { scrollVelocity, canvases, externalAccessService, accessToken } = data;
   const [mainLoaded, setMainLoaded] = useState(false);
   const currentCanvas = canvases[index];
-  const mainImageService = { '@id': currentCanvas.imageServiceId };
-  const urlTemplateMain = mainImageService['@id']
-    ? iiifImageTemplate(mainImageService['@id'])
+  const { user } = useUser();
```

Comment:
Not a real suggestion for this just a thought; was wondering if it was worth adding something like

`const isStaffWithRestricted = user?.role === 'StaffWithRestricted'`

in the `useUser()` context/hook. Since we do this check often!